lib/RecommendationProvider.js

```diff
@@ -0,0 +1,157 @@
+    this.getARecommendation(getRecommendation);
+  },
+
+  /**
+    * Randomly pick a recommendation from the allowed set of recommendations (i.e ones that are not blocked)
+    */
+  _getRandomRecommendation() {
+    const allowedRecommendations = this._recommendedContent.filter(item => !this._blockedRecommendedContent.includes(item.url));
+    if (allowedRecommendations.length > 0) {
+      let index = Math.floor(Math.random() * (allowedRecommendations.length));
+      this._currentRecommendation = Object.assign({}, {
+        url: allowedRecommendations[index].url,
+        timestamp: allowedRecommendations[index].timestamp,
+        recommended: true});
+      return this._currentRecommendation;
+    }
+    this._currentRecommendation = null;
+    return this._currentRecommendation;
+  },
+
+  /**
+    * Fetches Pocket recommendations and saves them to a global list of recommended content
+    */
+  asyncSetRecommendedContent: Task.async(function*() {
+    try {
+      let response = yield this._asyncGetRecommendationsFromPocket();
+      if (response.ok) {
+        let responseJson = yield response.json();
+        if (responseJson.urls.length > 0) {
+          // wait for the recommendation's metadata to come back (we don't want to show ugly recommendations)
+          let richRecommendations = yield this._previewProvider.getLinkMetadata(responseJson.urls);
+          this._recommendedContent = richRecommendations;
+          this._recommendedContent.forEach(item => item.timestamp = this._convertTime(item.timestamp));
+        }
+      }
+    } catch (err) {
+      Cu.reportError(err);
+      throw err;
+    }
+  }),
+
+  /**
+    * Update the list of recommendations the user has blocked (they do not want to see these again)
+    */
+  setBlockedRecommendation(link) {
+    this._blockedRecommendedContent.push(link);
+    this._currentRecommendation = null;
+  },
+
+  /**
+    * Convert milliseconds into human readable form and style it accordingly
+    */
+  _convertTime(timestamp) {
+    let publishedTime = Date.now() - timestamp;
+    if (timestamp === 0 || publishedTime < MILLISECONDS_IN_TEN_MINUTES) {
+      return "Now";
+    } else if (publishedTime >= MILLISECONDS_IN_TEN_MINUTES && publishedTime < MILLISECONDS_IN_AN_HOUR) {
+      return Math.floor(((publishedTime / (MILLISECONDS_IN_A_MINUTE)) % 60)) + "m";
+    } else if (publishedTime >= MILLISECONDS_IN_AN_HOUR && publishedTime < MILLISECONDS_IN_A_DAY) {
+      return Math.floor(((publishedTime / (MILLISECONDS_IN_AN_HOUR)) % 24)) + "h";
+    } else if (publishedTime >= MILLISECONDS_IN_A_DAY && publishedTime < MILLISECONDS_IN_A_WEEK) {
+      return Math.floor(((publishedTime / (MILLISECONDS_IN_A_DAY)) % 7)) + "d";
+    } else {
+      return Math.floor((publishedTime / (MILLISECONDS_IN_A_WEEK))) + "w";
+    }
+  },
+
+  _asyncGetRecommendationsFromPocket: Task.async(function*() {
+    try {
+      let response = yield fetch(this._pocketEndpoint, {
```

Comment:
There's no one using the RecommendationProvider just yet, so it will actually not ever even fire. I just removed the end point under Jared gives us a stage endpoint (the one that was there previously was just for testing purposes and shouldn't have made it to the PR)